client/hwlib/src/constants.rs

```diff
@@ -23,6 +23,9 @@ pub const PROC_CPUINFO_FILE_PATH: &str = "/proc/cpuinfo";
 pub const PROC_DEVICE_TREE_DIR_PATH: &str = "/proc/device-tree/";
 pub const PROC_VERSION_FILE_PATH: &str = "/proc/version";
 
+// The os-release file path is set during the build time
+// to allow for snap packaging, which exposes the file in a different location.
```

Comment:
I think the second line here is slightly leaky---there may be other reasons in future unrelated to snaps which require us to use `env!` which could cause this comment to quietly go out of date. I think it may be cleaner to just remove the second line here.

The first line is also slightly redundant, but given that this is a non-standard environment variable, I see no harm in keeping it. Small nitpick, I think it would be more natural to say that the '`os-release` file path is _defined_ in `build.rs`' (rather than 'set')